src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -292,6 +339,11 @@ def get_composite_package_identifier(self, package, package_version):
 
     def install_updates_fail_safe(self, excluded_packages):
         return
+
+    def install_security_updates_azgps_coordinated(self):
+        command = self.__generate_command(self.install_security_updates_azgps_coordinated_cmd, self.__get_custom_sources_to_spec(self.__get_custom_sources_to_spec(self.max_patch_publish_date, base_classification="security")))
```

Comment:
self.__get_custom_sources_to_spec() called twice? the outer one should throw an exception since self.custom_sources_list, returned by self.__get_custom_sources_to_spec(), is a path. Also self.install_security_updates_azgps_coordinated_cmd doesn't have a SOURCES reference

Reply:
Fixed - the code is set to gracefully degrade to install at least what's required (so doesn't throw). Great catch!

Reply:
Second question in my earlier comment: **Also self.install_security_updates_azgps_coordinated_cmd doesn't have a SOURCES reference**

Do we need to add SOURCES placeholder there for __generate_command() to replace?